core/src/main/java/hudson/security/HudsonPrivateSecurityRealm.java

```diff
@@ -358,7 +358,7 @@ private User createAccount(StaplerRequest req, StaplerResponse rsp, boolean self
         if(si.fullname==null || si.fullname.length()==0)
             si.fullname = si.username;
 
-        if(si.email==null || !si.email.contains("@"))
+        if(isMailerPluginPresent() && (si.email==null || !si.email.contains("@")))
```

Comment:
This is really crying out for an extension point—a way for `UserPropertyDescriptor` to indicate that it wishes to be configured during account creation, for security realms supporting such a concept.